cloudsim-plus/src/main/java/org/cloudbus/cloudsim/hosts/HostSimple.java

```diff
@@ -507,25 +517,46 @@ public boolean hasEverStarted() {
 
     @Override
     public final Host setActive(final boolean activate) {
-        if(this.active == activate){
+    	if(this.active == activate){
             return this;
         }
 
         if(isFailed() && activate){
             throw new IllegalStateException("The Host is failed and cannot be activated.");
         }
+        
+        double delay = activate? this.powerModel.getHostStartupDelay(): this.powerModel.getHostShutDownDelay();
```

Comment:
Define all variables as final whenever possible. Accuracy is a must in simulation. Taking these little measures avoids bugs / inconsistencies.